src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondExtensionCallbackHandler.java

```diff
@@ -0,0 +1,121 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.SubmittedCallbackResponse;
+import uk.gov.hmcts.reform.unspec.callback.Callback;
+import uk.gov.hmcts.reform.unspec.callback.CallbackHandler;
+import uk.gov.hmcts.reform.unspec.callback.CallbackParams;
+import uk.gov.hmcts.reform.unspec.callback.CallbackType;
+import uk.gov.hmcts.reform.unspec.callback.CaseEvent;
+import uk.gov.hmcts.reform.unspec.validation.RequestExtensionValidator;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static uk.gov.hmcts.reform.unspec.callback.CaseEvent.RESPOND_EXTENSION;
+import static uk.gov.hmcts.reform.unspec.helpers.DateFormatHelper.DATE;
+import static uk.gov.hmcts.reform.unspec.helpers.DateFormatHelper.formatLocalDateTime;
+
+@Service
+public class RespondExtensionCallbackHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = Collections.singletonList(RESPOND_EXTENSION);
+    public static final String COUNTER_DEADLINE = "respondentSolicitor1claimResponseExtensionCounterDate";
+    public static final String RESPONSE_DEADLINE = "responseDeadline";
+    public static final String PROPOSED_DEADLINE = "respondentSolicitor1claimResponseExtensionProposedDeadline";
+    public static final String EXTENSION_REASON = "respondentSolicitor1claimResponseExtensionReason";
+
+    private final ObjectMapper mapper;
+    private final RequestExtensionValidator validator;
+
+    public RespondExtensionCallbackHandler(ObjectMapper mapper, RequestExtensionValidator validator) {
+        this.mapper = mapper;
+        this.validator = validator;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return Map.of(
+            CallbackType.ABOUT_TO_START, this::prepopulateRequestReasonIfAbsent,
+            CallbackType.MID, this::validateRequestedDeadline,
+            CallbackType.ABOUT_TO_SUBMIT, this::updateResponseDeadline,
+            CallbackType.SUBMITTED, this::buildConfirmation
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse prepopulateRequestReasonIfAbsent(CallbackParams callbackParams) {
+        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
+        data.putIfAbsent(EXTENSION_REASON, "No reason given");
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(data)
+            .build();
+    }
+
+    private CallbackResponse validateRequestedDeadline(CallbackParams callbackParams) {
```

Comment:
If the user chooses "No" there won't be a extensionCounterDate to validate which will then cause validateProposedDeadline to return error "The proposed deadline must be after the current deadline" and this will be propagated back to the user due to dateToValidate == null